### PR TITLE
Add eslint infrastructure

### PR DESCRIPTION
**Pull request by Amir-61**

### CODE REVIEW

File: test/migration.test.js [hunk 1]
```diff
@@ -65,16 +64,17 @@ describe('migrations', function () {
...
     });
   });
 
-  it('UserData should have correct indexes', function (done) {
+  it('UserData should have correct indexes', function(done) {
     // Note: getIndexes truncates multi-key indexes to the first member.
     // Hence index1 is correct.
-    getIndexes('UserData', function (err, fields) {
+    getIndexes('UserData', function(err, fields) {
+      /* eslint-disable camelcase */
```
Comment (Amir-61) [inline test/migration.test.js]: Please note: I disabled camelcase checks here, since it is a breaking change.

File: lib/enumFactory.js [hunk 2]
```diff
@@ -23,26 +23,26 @@ var EnumFactory = function() {
...
       enumerable: false,
       value: stringified(Enum),
-      writable: false
+      writable: false,
     });
     Object.freeze(Enum);
     return Enum;
   } else {
-    throw "No arguments - could not create Enum.";
+    throw 'No arguments - could not create Enum.';
```
Comment (superkhau) [inline lib/enumFactory.js]: we should change this to `throw new Error('No arguments - count not create Enum.'`. Please commit a separate PR for this.

File: test/mysql.autoupdate.test.js [hunk 3]
```diff
@@ -2,93 +2,93 @@ var assert = require('assert');
...
-          "type": "String",
-          "length": 20,
-          "id": 1
+      {
+        'name': 'CustomerTest',
+        'options': {
+          'idInjection': false,
+          'mysql': {
+          'schema': 'myapp_test',
+          'table': 'customer_test',
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: remove single quotes from keys

File: test/mysql.autoupdate.test.js [hunk 4]
```diff
@@ -2,93 +2,93 @@ var assert = require('assert');
...
         },
-        "email": {
-          "type": "String",
-          "required": true,
-          "length": 40
+        'properties': {
+          'id': {
+          'type': 'String',
+          'length': 20,
+          'id': 1,
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: ditto

File: test/mysql.autoupdate.test.js [hunk 5]
```diff
@@ -2,93 +2,93 @@ var assert = require('assert');
...
+          'type': 'String',
+          'required': true,
+          'length': 40,
+        },
+          'age': {
+          'type': 'Number',
+          'required': false,
+        },
+        },
+      };
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: ditto

File: test/mysql.autoupdate.test.js [hunk 6]
```diff
@@ -2,93 +2,93 @@ var assert = require('assert');
...
+            'columnName': 'email',
+            'dataType': 'varchar',
+            'dataLength': 60,
+            'nullable': 'YES',
+          },
+        },
+          'firstName': {
+          'type': 'String',
+          'required': false,
+          'length': 40,
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: ditto

File: test/mysql.autoupdate.test.js [hunk 7]
```diff
@@ -2,93 +2,93 @@ var assert = require('assert');
...
-          "mysql": {
-            "columnName": "email",
-            "dataType": "varchar",
-            "dataLength": 60,
-            "nullable": "YES"
-          }
+          'lastName': {
+          'type': 'String',
+          'required': false,
+          'length': 40,
```
Comment (superkhau) [inline test/mysql.autoupdate.test.js]: ditto

File: test/mysql.discover.test.js [hunk 8]
```diff
@@ -227,9 +227,9 @@ describe('Discover LDL schema from a table', function () {
   });
 });
 
-describe('Discover and build models', function () {
-  it('should discover and build models', function (done) {
-    db.discoverAndBuildModels('INVENTORY', {owner: 'STRONGLOOP', visited: {}, associations: true}, function (err, models) {
+describe('Discover and build models', function() {
+  it('should discover and build models', function(done) {
+    db.discoverAndBuildModels('INVENTORY', { owner: 'STRONGLOOP', visited: {}, associations: true }, function(err, models) {
```
Comment (superkhau) [inline test/mysql.discover.test.js]: this is a long line, i'm suprised eslint isn't enforcing 80 char limit

**Comment by superkhau** [comment]
Minor comments, please fix, and squash. No more review is needed for this one after, LGTM.

**Comment by Amir-61** [comment]
Closing this PR; code has changed and eslint fix in not valid anymore.

**Pull request closed without merging**